### ISSUE TITLE
Use Symfony DI container instead of service factories

### ISSUE THREAD

**Issue by phavekes** [OpenConext/OpenConext-engineblock#1762]
[This issue is imported from pivotal - Originaly created at Jan 24, 2017 by Alex Rothuis](https://www.pivotaltracker.com/story/show/138225085)

Currently there are some services that are being built in [EngineBlock_Corto_Module_Services](https://github.com/OpenConext/OpenConext-engineblock/blob/46ff3d6826dad9267c6868285731b2c37c45d33a/library/EngineBlock/Corto/Module/Services.php#L65-L95). 

EngineBlock Corto services tend to extend `EngineBlock_Corto_Module_Service_Abstract`, services created using a factory do not extend this class because they have their own special needs. By using the Symfony DI container and YML files, each service can be configured in their own specific way.

**Comment by phavekes** [comment]
Already fixed this in 5.7? (Michiel Kodde - Jul 2, 2018)

**Comment by phavekes** [comment]
Making Corto use Symfony-DI 100% is not doable. The mentioned case could be fixed. Todo: see what can be done / what the benefits are. (bstrooband - Jul 19, 2018)

**Comment by johanib** [comment]
In the meantime, we implemented a new DiContainer that is populated using the Symfony service container.

**Migrating services from DiContainer to DiContainerRuntime**
90% of the services in the DiContainer can be moved over to the DiRuntimeContainer so we can use the Symfony DI container instead of the old Pimple container. This way, Symfony can optimize the service container and services are no longer plucked from the container at runtime, which is deprecated. 

But not everything. For example: \OpenConext\EngineBlockBundle\OpenConextEngineBlockBundle::boot
During the boot of the EngineBlockBundle the DiContainer is used. But at that time the Symfony Container is not available yet. So we leave those as is.

See https://github.com/OpenConext/OpenConext-engineblock/blob/18e5fa7f9ddf0b81ef62e5669200ac76caa170bf/library/EngineBlock/Application/DiContainer.php#L39 

See https://github.com/OpenConext/OpenConext-engineblock/blob/45bd19f5b38bdd8c682918bb4f09954a438f73a4/src/OpenConext/EngineBlockBundle/Bridge/DiContainerRuntime.php#L29


**Migrating services from DiContainer > Symfony services**
Instead of moving everything to the DiContainerRuntime, it might be possible to skip that step for callers outside of corto:
 1. src/ callers > inject directly, no intermediary
 2. library/ callers > migrate to DiContainerRuntime

**Corto service factories**
With regards to the Service Factories in corto, that seems to be a different case.
But I think that it's not really usefull to refactor the `serviceFactory` at this point, because it's deep inside Corto. Migrating services to the Symfony DI container is a first step.

See https://github.com/OpenConext/OpenConext-engineblock/blob/45bd19f5b38bdd8c682918bb4f09954a438f73a4/library/EngineBlock/Corto/Module/Services.php#L75-L92


**tl;dr**
Move services from DiContainer to DiContainerRuntime. This is a low cost operation and a small step toward removing the DiContainer.

**Comment by johanib** [comment]
Discussion in refinement: Not for now. The gains are minimal, better to refactor when we touch the actual code.
